framework/src/userobject/InterfaceUserObject.C

```diff
@@ -0,0 +1,61 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "InterfaceUserObject.h"
+#include "SubProblem.h"
+#include "MooseTypes.h"
+#include "Assembly.h"
+
+template <>
+InputParameters
+validParams<InterfaceUserObject>()
+{
+  InputParameters params = validParams<UserObject>();
+  params += validParams<BoundaryRestrictableRequired>();
+  params += validParams<TwoMaterialPropertyInterface>();
+  params += validParams<TransientInterface>();
+  params.addClassDescription("Basic UO class to perform computation across an interface");
+  return params;
+}
+
+InterfaceUserObject::InterfaceUserObject(const InputParameters & parameters)
+  : UserObject(parameters),
+    BoundaryRestrictableRequired(this, false), // false for applying to sidesets
+    TwoMaterialPropertyInterface(this, Moose::EMPTY_BLOCK_IDS, boundaryIDs()),
+    NeighborCoupleable(this, false, false),
+    MooseVariableDependencyInterface(),
+    UserObjectInterface(this),
+    TransientInterface(this),
+    PostprocessorInterface(this),
+    _mesh(_subproblem.mesh()),
+    _q_point(_assembly.qPointsFace()),
+    _qrule(_assembly.qRuleFace()),
+    _JxW(_assembly.JxWFace()),
+    _JxW_neighbor(_assembly.JxWNeighbor()),
```

Comment:
I would like to give access to this user object the `_JxW_neighbor`. However if I use _JxW_neighbor[qp] in exectue I receive an assertion error:

```
Assertion `i < _size' failed
Access out of bounds in MooseArray (i: 0 size: 0)
at /home/arovinelli/projects/moose/framework/build/header_symlinks/MooseArray.h, line 268
Detaching after fork from child process 6128.
Stack frames: 15
0: libMesh::print_trace(std::ostream&)
1: MooseArray<double>::operator[](unsigned int) const
2: InterfaceUO::execute()
3: ComputeUserObjectsThread::onInterface(libMesh::Elem const*, unsigned int, short)
4: ThreadedElementLoopBase<libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> >::operator()(libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> const&, bool)
5: void libMesh::Threads::parallel_reduce<libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*>, ComputeUserObjectsThread>(libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> const&, ComputeUserObjectsThread&)
6: FEProblemBase::computeUserObjects(MooseEnumItem const&, Moose::AuxGroup const&)
7: FEProblemBase::execute(MooseEnumItem const&)
8: FEProblemBase::initialSetup()
9: Steady::init()
10: MooseApp::executeExecutioner()
11: MooseApp::run()
12: /home/arovinelli/projects/moose/test/tests/userobjects/interface_user_object/../../../moose_test-dbg() [0x40e818]
13: __libc_start_main
14: /home/arovinelli/projects/moose/test/tests/userobjects/interface_user_object/../../../moose_test-dbg() [0x40e529]
[0] /home/arovinelli/projects/moose/framework/build/header_symlinks/MooseArray.h, line 268, compiled Feb 15 2019 at 08:30:52
```

Am I missing something to reinit something in the loops  or simply  this not supposed to be used?

Reply:
@jwpeterson any idea about this?

Reply:
For now, I removed access to _JxW_neighbor. This implies small deformations for a disconnect mesh (e.g when using `BreakMeshByBlock`).